### ISSUE TITLE
Skipping Attack / Refreshing?

### ISSUE THREAD

**Issue by pannatatm** [Frizz925/gbf-autopilot-core#14]
I'm trying to refresh the page after attack but after refresh bot not reloading the script and do another attack. If I put SkipAttack() at the last line of scripts the bot should check from beginning again but it seem GBFAutopilot do not recognize this command yet.

This is script I'm trying to use

`EnableChargeAttack()
Summon(6)

if character_1.name == "MC" then
  character_1:UseSkill(4)
  character_1:UseSkill(3)
  character_1:UseSkill(1)
end

if characters["Zooey"] ~= nil then
  characters["Zooey"]:UseSkill(2)
  characters["Zooey"]:UseSkill(1)
  characters["Zooey"]:UseSkill(3)
end

if characters["Jeanne d'Arc"] ~= nil then
  characters["Jeanne d'Arc"]:UseSkill(2)
  characters["Jeanne d'Arc"]:UseSkill(1)
end

if characters["Narmaya"] ~= nil then
  characters["Narmaya"]:UseSkill(2)
  characters["Narmaya"]:UseSkill(1):WithWaitTime(3000)
end

if (turn > 1 and character_1.name == "MC") then
  character_1:UseSkill(2)
end

if (turn > 1 and characters["Jeanne d'Arc"] ~= nil) then
  characters["Jeanne d'Arc"]:UseSkill(3)
end

if (turn > 1 and characters["Narmaya"] ~= nil) then
  characters["Narmaya"]:UseSkill(3)
end

Attack() // << The bot do normal attack
Refresh() // << Then it refresh but after refresh bot continue to do another attack eventhough it should be checking turn and using skill that set to be "turn > 1"
Wait(3000)
SkipAttack() // << This function should fix the problem but it seem not implemented in Autopilot yet
`

My current solution is hotfix by putting code I want to use on turn 2 after Refresh() and write another Attack() and Refresh() line. But for long-term time wise I wish the SkipAttack() function work or somekind of built-in auto reloading after attack implement like in Z bot.

**Comment by pannatatm** [comment]
This can be low priority as it fix easily by using "goto" in lua script. However I found another problem being the bot cannot detect enemy state after refreshing the page. Bot seem to not recognize enemy overdrive/break state.

**Comment by Frizz925** [comment]
There's been many requests for this feature actually. But to implement the attack refresh, problem from #1 must be solved entirely first, mainly because refreshing too early causes "processing for last turn" popup and causes the bot to get stuck (and replicating the popup is kind of random on my side so it's kind of difficult).

I can try implement an early version of this feature but its usage must be taken with care to not cause the bot gets stuck very often.

**Comment by pannatatm** [comment]
Yea I'm manually detect the ougi bar and set wait time. Run pretty smooth.

```
WaitAfterRefresh = 0
::start::
if character_1.charge_gauge == 100 then
  WaitAfterRefresh = 8000
  else
  WaitAfterRefresh = 1000
  DisableChargeAttack()
end

Wait(500)
Attack()
Wait(500)
Refresh()
Wait(WaitAfterRefresh)
goto start
```

I thinking of writing longer code to detect every possiblity of ougi chain count and adjust wait time for it. But the solution I'm thinking is quite too many line of code. (since my programming skill is quite limited :p) I think there should be better way to detect ougi chain before clicking attack that return value of chain number like>

if party.chain_burst = 1 then
  WaitAfterRefresh = 4000
  else if party.chain_burst = 2 then
  WaitAfterRefresh = 7000
  else if party.chain_burst = 3 then
  WaitAfterRefresh = 9000
  else if party.chain_burst = 4 then
  WaitAfterRefresh = 13000
else
  WaitAfterRefresh = 1000
end

From what I experienced single attack full party cause 4 second server-side. 
And +0.5 for each multi attack party member did.
So full party triple attack will take around 7-8 second server-side.
And ougi take around 3 second for each party member
So full chain = 12 second + (base delay 1 second) = 13 second.
Hope that can help you in someway :)

**Comment by Frizz925** [comment]
Good idea. But there's another caveat if any party member has an uplift or CA seal effect. The bot could've just counted the charge gauge of every party member and predict how many chain bursts it would occur. This works flawlessly, however, if the mentioned buffs/debuffs aren't in effect.

**Comment by Homusamu** [comment]
User can just disable auto refresh on fight where enemy inflict seal/slow a lot but in most of fight this logic should work perfectly. especially for most of short farming content could benefit from it a lot. I find myself saving almost half of the time compared to no refresh.

And for more advance method bot can even detect CA seal on character and calculate according to it. Still in some case like Qilin Harp where ougi can cure party member status effect will mess with it tho (lol) so I think just simple basic SkipAttack() and party.chain_burst that return value without other factor is enough for now. User can manually set wait time delay themselves depends on battle too.
